theia-ide: init at 1.63.200

goyalyashpal opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Theia-ide is an Open source, vendor neutral IDE for cloud & desktop, built on Theia Platform.
See popularity info & links to homepages at: 
* https://github.com/NixOS/nixpkgs/issues/323316
* https://github.com/NixOS/nixpkgs/issues/323316#issuecomment-2926856945

Closes: https://github.com/NixOS/nixpkgs/issues/323316

----
i will post the progress report on this in 2nd comment below.

i am thinking of doing squashing commits & force-pushing only at the end, once everything has been reviewed thoroughly on the source side.

----
## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux --> DRAFT
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md), [maintainers/README.md](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) and other contributing documentation in corresponding paths.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

goyalyashpal committed:
theia-ide: init at 1.62.200

goyalyashpal committed:
maintainers: add goyalyashpal

acid-bong reviewed (commented):
A couple of notes. The rest looks fine so far

goyalyashpal commented on pkgs/by-name/th/theia/package.nix:
```diff
@@ -0,0 +1,305 @@
+    ## Needed for Compiling the native modules
+    pkg-config
+    libsecret
+    xorg.libxkbfile
+    xorg.libX11
```
> they don't belong in `nativeBuildInputs`. Consider moving to `buildInputs`

hi!  thanks for taking a look.

i tried removing these & the process fails at `buildPhase` with following error:

```console
$ nix build   # in theia dir using flake.nix
$ nix log /nix/store/...-theia-ide-...drv
⠇ Building module: native-keymap, Completed: 3
Package xkbfile was not found in the pkg-config search path.
Perhaps you should add the directory containing `xkbfile.pc'
to the PKG_CONFIG_PATH environment variable
No package 'xkbfile' found
gyp: Call to '${PKG_CONFIG:-pkg-config} x11 xkbfile --libs' returned exit status 1 while in binding.gyp. while trying to load binding.gyp
✖ Rebuild Failed

An unhandled error occurred inside electron-rebuild
node-gyp failed to rebuild '/build/source/node_modules/native-keymap'.
For more information, rerun with the DEBUG environment variable set to "electron-rebuild".

Error: `gyp` failed with exit code: 1

```

goyalyashpal committed:
enh(inputs): Use top-level X-libs ...

* This is partial (incomplete) commit.

Co-authored-by: Acid Bong <acidbong@tilde.club>

goyalyashpal committed:
enh(inputs): Add left X-libs

goyalyashpal committed:
refac(env): Move DEBUG var up one line

goyalyashpal committed:
enh(src): Use fetchFromGitHub.tag for version instead of rev

goyalyashpal committed:
docs: Add TODO to parametrise fetch nodeHeaders

goyalyashpal committed:
enh(nodeheaders): Set nodedir to nodejs

goyalyashpal committed:
docs: Improve comments on pipe, map functions

goyalyashpal committed:
refac: mv theia{,-ide}

goyalyashpal committed:
docs: Add searchable comment for nodedir env-var

goyalyashpal committed:
fix(meta): Correct typo in license

goyalyashpal committed:
build(nodejs): Use nodejs v22 ...

  * https://search.nixos.org/packages?&show=electron&channel=unstable
  * https://releases.electronjs.org/release/v35.6.0

goyalyashpal committed:
style: `nix fmt .../theia-ide/package.nix`

goyalyashpal committed:
docs: Add headers & errors to explain reason

goyalyashpal committed:
enh: Remove libdrm, Use cups.lib instead of just cups

goyalyashpal committed:
docs: Reason for cups.lib

goyalyashpal committed:
docs: Remove redundant line

goyalyashpal committed:
enh: Add --dir option explicitly

goyalyashpal committed:
refac: Use full options for readability

goyalyashpal committed:
refac: Use full option for readability (install)

goyalyashpal committed:
trial: Use arch theia-electron PKGBUILD package()

goyalyashpal committed:
trial: usr/lib/$pname -> opt/$pname per archlinux/electron_pkg_guide

goyalyashpal committed:
refac: Move autoPatchelfHook to distribution

goyalyashpal committed:
feat: Enable ch{own,mod} on chrome-sandbox

goyalyashpal committed:
feat: Write installPhase related to $out/bin/

goyalyashpal committed:
feat: Remove ch{own,mod} and unneeded makeWrapper code comment

goyalyashpal committed:
fix: Replace all instances of pname with custom variable

goyalyashpal committed:
theia-ide: from 1.62.200 -> 1.63.200

goyalyashpal committed:
build: Use electron_36

*
https://github.com/eclipse-theia/theia/commit/e9b853ea9ccf58dfa74f8204240f3238aa832f6a

goyalyashpal commented:
closing as it is not a "work in progress", which is what this type of PRs used to mean back when they were called "WIP: ".

though "draft" term suits the current situation, but the "open" status may give someone false hope.

feel free to reopen if you can help resolve the errors.
